manifest.go

```diff
@@ -460,12 +460,12 @@ type hasFieldToIDMap interface {
 	setFieldIDToFixedSizeMap(map[int]int)
 }
 
-func fetchManifestEntries(m ManifestFile, fs iceio.IO, discardDeleted bool) ([]ManifestEntry, error) {
+func fetchManifestEntries(m ManifestFile, fs iceio.IO, discardDeleted bool) (_ []ManifestEntry, err error) {
 	f, err := fs.Open(m.FilePath())
 	if err != nil {
 		return nil, err
 	}
-	defer f.Close()
+	defer internal.CheckedClose(f, &err)
 
 	return ReadManifest(m, f, discardDeleted)
```

Comment:
this won't work the way you intend since we are returning whatever error is returned by `ReadManifest`. Instead you would do something like this:

```go
func fetchManifestEntries(m ManifestFile, fs iceio.IO, discardDeleted bool) ([]ManifestEntry, error) {
    f, err := fs.Open(m.FilePath())
    if err != nil {
        return nil, err
    }

    defer internal.CheckedClose(f, &err)

    var entries []ManifestEntry
    entries, err = ReadManifest(m, f, discardDeleted)
    return entries, err
}
```

Reply:
oh, good catch, thanks!

Reply:
checked this, looks like we're fine returning other things than err (https://go.dev/play/p/rj3gfSL_FiE)